### DISCUSSION TITLE
Building a Release build vs Debug build

### DISCUSSION THREAD

**Discussion by sbonar** [libuv/libuv#4061] [General]
In compiling the UV under Windows is see that it puts the output under the Debug directory.  It's been my experience that this typically means that the library has been built with little optimization to better help in debugging issues.
My question then is there a cmake flag that needs to be set in order to do a Release build that has more optimization enabled for performance or is the build system made so that debug and release are essentially the same?
TIA

**Comment by bnoordhuis** [comment]
There's a flag (`-DCMAKE_BUILD_TYPE=Release` - standard cmake flag) but libuv doesn't do much fancy math or algorithms, just i/o, so the performance of debug and release builds is usually very similar.

**Reply by sbonar** [reply]
I tried that flag and at least on Windows I got the message that the flag was not used by the project which prompted the question.

**Reply by hasansultan92** [reply]
it most definitely is being used. Were you able to figure this out @sbonar?
